### PR TITLE
Generate code into OUT_DIR.

### PR DESCRIPTION
**Pull request by metajack**
This is necessary to ensure Cargo knows when to rebuild. Normally
.gitignore would be enough to exclude these from Cargo's freshness
calculation, but https://github.com/rust-lang/cargo/issues/1729 prevents
this currently. This is the new, correct way to do these thigns, just
like the style crate does.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6408)

### CODE REVIEW

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -1429,25 +1429,26 @@ class CGImports(CGWrapper):
...
-        ignored_warnings = [
-            # Allow unreachable_code because we use 'break' in a way that
-            # sometimes produces two 'break's in a row. See for example
-            # CallbackMember.getArgConversions.
+        if ignored_warnings is None:
+            ignored_warnings = [
+                # Allow unreachable_code because we use 'break' in a way that
+                # sometimes produces two 'break's in a row. See for example
+                # CallbackMember.getArgConversions.
             'unreachable_code',
```
Comment (SimonSapin) [inline components/script/dom/bindings/codegen/CodegenRust.py]: nit: indentation

r=me with this

**Comment by Ms2ger** [comment]
For future reference, I really don't like having unrelated changes like the `ignored_warnings` changes here in other PRs

**Comment by metajack** [comment]
It wasn't unrelated at all. It was part of turning #! Attributes to # ones since the former now doesn't compile since the modules are included.

**Comment by SimonSapin** [comment]
One side effect of this is that doc.servo.org doesn’t have source links for generated code anymore. For example, the [src] link on http://doc.servo.org/script/dom/bindings/codegen/Bindings/DOMParserBinding/enum.SupportedType.html goes to http://doc.servo.org/src/script///home/servo/buildbot/slave/doc/build/target/debug/build/script-49135cd7c70c0a0f/out/Bindings/DOMParserBinding.rs.html#99-102 which is a 404.

**Comment by Ms2ger** [comment]
Ugh, that was one of our big advantages over Gecko...

**Comment by metajack** [comment]
I think we can consider this a bug in rustdoc. Can someone file this?

**Comment by SimonSapin** [comment]
https://github.com/rust-lang/rust/issues/26995

**Pull request merged**